pandas/_libs/tslibs/strptime.pyx

```diff
@@ -60,15 +62,23 @@ cdef bint format_is_iso(f: str):
     Generally of form YYYY-MM-DDTHH:MM:SS - date separator can be different
     but must be consistent.  Leading 0s in dates and times are optional.
     """
+    iso_regex = re.compile(
+        r"""
+        ^                     # start of string
+        %Y                    # Year
+        (?:([-/ \\.]?)%m      # month with or without separators
+        (?: \1%d              # day with same separator as for year-month
+        (?:[ |T]%H            # hour with separator
```

Comment:
square brackets match anything inside them, you don't need `|`
```suggestion
        (?:[ T]%H             # hour with separator
```

Reply:
should I directly commit the suggestions?